lib/Mojo/Base.pm

```diff
@@ -101,6 +105,16 @@ sub tap {
   return $self;
 }
 
+sub with_roles {
+  if ($_[0]->can_roles()) {
```

Comment:
Within the class, it should use the constant (see above), since this will be constant folded at compile time and not introduce overhead on each with_roles call